lib/manager/components/LabelEditorModal.js

```diff
@@ -0,0 +1,69 @@
+import React from 'react'
+import { Modal } from 'react-bootstrap'
+
+// @flow
+
+import LabelEditor from '../components/LabelEditor'
+import type { Label } from '../../types'
+
+type Props = {
+  label?: Label,
+  projectId?: String
+}
+
+type State = {
+  showModal: boolean
+}
+
+/**
+ * Renders a LabelEditor within a Modal, including a dynamic title depending
+ * on if a label is new
+ */
+export default class LabelEditorModal extends React.Component<
+  Props,
+  State
+> {
+  state = {
+    showModal: false
+  };
+
+  close = () => {
+    this.setState({
+      showModal: false
+    })
+  };
+
+  // Used in the ref
```

Comment:
I can't recall a recent extensive use of refs like this for this repo... Is there an alternate approach?

Reply:
I based the Modal files on https://github.com/ibi-group/datatools-ui/blob/dev/lib/common/components/InfoModal.js

My original approach used React hooks, which aren't available in the version of React we're using!